app/templates/chat.hbs

```diff
@@ -1,9 +1,46 @@
 <header>
-  <span class="chat-title ">Chat</span>
-  <div class="search-div">
-    {{search-bar placeholder="Search Messages"}}
-    <img src="https://png.icons8.com/search-filled/ios7/39/F9C20A">
-  </div>
+  {{#if (eq appController.currentRouteName "chat.chat")}}
+      {{top-header
+        title="Chat"
+        clickFilter=(action chatController.actions.onClickFilter)
+      }}
+      {{search-bar
+        placeholder=chatController.searchBarText
+        onInput=(action chatController.actions.onSearchBarInput)
+        value=chatController.searchValue
```

Comment:
Looks good, just one question. Where is the searchValue defined in each controller, is it inherited from somewhere?

Reply:
I never declared searchValue explicitly, but what it does is it takes the value from the search bar and saves it to "searchValue" in the controller.